main.go

```diff
@@ -62,6 +61,30 @@ func run() error {
 		return errors.Wrap(err, "unable to get global arguments")
 	}
 
+	if globalArgs.LogFileDir != "" {
```

Comment:
Can you extract line 64-84 into a separate function so that the main logic would look simpler?